### PR TITLE
Add test cases for HtmlReportBuilder

### PR DESCRIPTION
**Pull request by djydewang**
The output of HtmlReportBuilder in test cases:

- [spring-framework-diff-report](https://djydewang.github.io/spring-framework-test/index.html)
   The pmd reports of spring-framework is artificial. I intercepted part of the normal pmd report and made a retrofit. And the diff report was builded according to the pmd reports mentioned before.

- [empty report](https://djydewang.github.io/openjdk10/index.html)

### CODE REVIEW

**Comment by djydewang** [comment]
The new diff [report](https://djydewang.github.io/spring-framework-new/index.html).

File: test/test_html_report_builder.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    PmdTester::HtmlReportBuilder.new.build(project, report_diffs)
+
+    # Checking  css resources are copied into the diff report directory.
+    assert_equal(true, File.exist?("#{css_path}/maven-base.css"))
+    assert_equal(true, File.exist?("#{css_path}/maven-theme.css"))
+
+    # Checking the content of diff report is expected.
+    expected_file = File.open(EXPECTED_REPORT_PATH).read
+    actual_file = File.open("#{actual_report_path}/index.html").read
+    assert_equal(expected_file, actual_file)
```
[the remark this answers is not in the document]
Comment (djydewang) [reply test/test_html_report_builder.rb]: > How does the ruby test report the failure?

Ruby test would output the entire contents of both files in this case.

File: test/test_html_report_builder.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  ORIGINAL_BASE_PMD_REPORT_PATH = 'test/resources/test_html_report_builder_base.xml'
+  ORIGINAL_PATCH_PMD_REPORT_PATH = 'test/resources/test_html_report_builder_patch.xml'
+
+  TARGET_TEST_RESOURCES_PATH = 'target/test/resources'
+  BASE_PMD_REPORT_PATH = "#{TARGET_TEST_RESOURCES_PATH}/test_html_report_builder_base.xml"
+  PATCH_PMD_REPORT_PATH = "#{TARGET_TEST_RESOURCES_PATH}/test_html_report_builder_patch.xml"
+
+  EXPECTED_REPORT_PATH = 'test/resources/index.html'
+  EXPECTED_EMPTY_REPORT_PATH = 'test/resources/empty.html'
```
[the remark this answers is not in the document]
Comment (djydewang) [reply test/test_html_report_builder.rb]: I originally had another idea to create a report_diffs object based on a diff-report.xml file.
But such an approach, such as reading&parsing diff-report.xml etc., involves code similar to that of DiffBuilder.

File: test/test_html_report_builder.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    project = PmdTester::ProjectsParser.new.parse('test/resources/project-list.xml')[2]
+
+    actual_report_path = "target/reports/diff/#{project.name}"
+    css_path = "#{actual_report_path}/css"
+
+    build_pmd_report(ORIGINAL_BASE_PMD_REPORT_PATH, BASE_PMD_REPORT_PATH)
+    build_pmd_report(ORIGINAL_PATCH_PMD_REPORT_PATH, PATCH_PMD_REPORT_PATH)
+
+    diff_builder = PmdTester::DiffBuilder.new
+    report_diffs = diff_builder.build(BASE_PMD_REPORT_PATH, PATCH_PMD_REPORT_PATH)
```
[the remark this answers is not in the document]
Comment (djydewang) [reply test/test_html_report_builder.rb]: I originally had another idea to create a report_diffs object based on a diff-report.xml file.
But such an approach, such as reading&parsing diff-report.xml, etc., involves code similar to that of DiffBuilder.

File: test/test_html_report_builder.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (adangel) [reply test/test_html_report_builder.rb]: I hoped, it would do something like a diff automatically...

**Pull request merged**